pythclient/price_feeds.py

```diff
@@ -0,0 +1,430 @@
+        signature = vaa[start + 1 : start + 66]
+        guardian_signatures.append({"index": index, "signature": signature})
+
+    body = vaa[sig_start + sig_length * num_signers :]
+    version = vaa[0]
+    guardian_set_index = unpack(">I", vaa[1:5])[0]
+    timestamp = unpack(">I", body[0:4])[0]
+    nonce = unpack(">I", body[4:8])[0]
+    emitter_chain = unpack(">H", body[8:10])[0]
+    emitter_address = body[10:42]
+    sequence = int.from_bytes(body[42:50], byteorder="big")
+    consistency_level = body[50]
+    payload = body[51:]
+
+    # Compute the hash using pycryptodome's keccak
+    keccak_hash = keccak.new(digest_bits=256)
+    keccak_hash.update(body)
+    hash_val = keccak_hash.hexdigest()
+
+    return {
+        "version": version,
+        "guardian_set_index": guardian_set_index,
+        "guardian_signatures": guardian_signatures,
+        "timestamp": timestamp,
+        "nonce": nonce,
+        "emitter_chain": emitter_chain,
+        "emitter_address": emitter_address,
+        "sequence": sequence,
+        "consistency_level": consistency_level,
+        "payload": payload,
+        "hash": hash_val,
+    }
+
+
+def parse_batch_price_attestation(bytes_):
+    offset = 0
+
+    magic = bytes_[offset : offset + 4].decode("utf-8")
+    offset += 4
+    if magic != P2W_FORMAT_MAGIC:
+        raise ValueError(f"Invalid magic: {magic}, expected: {P2W_FORMAT_MAGIC}")
+
+    version_major = int.from_bytes(bytes_[offset : offset + 2], byteorder="big")
+    offset += 2
+    if version_major != P2W_FORMAT_VER_MAJOR:
+        raise ValueError(
+            f"Unsupported major version: {version_major}, expected: {P2W_FORMAT_VER_MAJOR}"
+        )
+
+    version_minor = int.from_bytes(bytes_[offset : offset + 2], byteorder="big")
+    offset += 2
+    if version_minor < P2W_FORMAT_VER_MINOR:
+        raise ValueError(
+            f"Unsupported minor version: {version_minor}, expected: {P2W_FORMAT_VER_MINOR}"
+        )
+
+    header_size = int.from_bytes(bytes_[offset : offset + 2], byteorder="big")
+    offset += 2
+    header_offset = 0
+
+    payload_id = int.from_bytes(
+        bytes_[offset + header_offset : offset + header_offset + 1], byteorder="big"
+    )
+    header_offset += 1
+    if payload_id != P2W_FORMAT_PAYLOAD_ID:
+        raise ValueError(
+            f"Invalid payload_id: {payload_id}, expected: {P2W_FORMAT_PAYLOAD_ID}"
+        )
+
+    offset += header_size
+    batch_len = int.from_bytes(bytes_[offset : offset + 2], byteorder="big")
+    offset += 2
+    attestation_size = int.from_bytes(bytes_[offset : offset + 2], byteorder="big")
+    offset += 2
+
+    price_attestations = []
+    for i in range(batch_len):
+        price_attestations.append(
+            parse_price_attestation(bytes_[offset : offset + attestation_size])
+        )
+        offset += attestation_size
+
+    if offset != len(bytes_):
+        raise ValueError(f"Invalid length: {len(bytes_)}, expected: {offset}")
+
+    return {
```

Comment:
why does this return a dictionary of one element?

Reply:
which one are you referring to?

Reply:
I followed the implementation here: https://github.com/pyth-network/pyth-crosschain/blob/main/wormhole_attester/sdk/js/src/index.ts#L122-L188

Reply:
oh... maybe leave a comment to that effect? I'm not sure why that function does it that way either.

Reply:
This is because that the returned struct was BatchPriceAttestation